test/lib/next-modes/next-dev.ts

```diff
@@ -167,17 +167,19 @@ export class NextDevInstance extends NextInstance {
         if (newFile) {
           await this.handleDevWatchDelayAfterChange(filename)
         } else if (filename.startsWith('next.config')) {
-          if (!this.cliOutput.slice(cliOutputLength).includes('Ready in')) {
-            throw new Error('Server has not finished restarting.')
-          }
+          await retry(async () => {
+            if (!this.cliOutput.slice(cliOutputLength).includes('Ready in')) {
```

Comment:
not directly related to your changes, but it's tricky when we have all these magic "Ready in" strings everywhere. Maybe worth at least refactoring to use [`this.serverReadyPattern`](https://github.com/vercel/next.js/blob/canary/test/lib/next-modes/base.ts#L72)

Reply:
Good point! Seems like I introduced `serverReadyPattern` in #67767 after #67767. Will fix...